src/components/graph/graph.helper.js

```diff
@@ -313,7 +313,9 @@ function buildNodeProps(node, config, nodeCallbacks = {}, highlightedNode, highl
         stroke,
         strokeWidth: strokeWidth * t,
         svg,
-        type: node.symbolType || config.node.symbolType
+        type: node.symbolType || config.node.symbolType,
+        data: node,
+        viewGenerator: node.viewGenerator || (!node.svg ? config.node.viewGenerator : null)
```

Comment:
To keep a more consistent functionality I think we should just go with:

```javascript
viewGenerator: node.viewGenerator || config.node.viewGenerator
```

Reply:
Before changing this, I'd like to explain why I did it like that. Once you know that, let me know if you'd still like me to go on with your proposal. 👍  

```node.viewGenerator || (!node.svg ? config.node.viewGenerator : null)```

With this, what I wanted to do was to stablish some priorities to decide how to render the node, just in case a single node has a value for the `svg` property .

**If a node has a `viewGenerator` at node level, the custom view will always be applied**, whether it also has a value for the `svg` property or not.

However, **if there's a `viewGenerator` at the global config but the node has a `svg` at node level, the `svg` should take precedence to the default `viewGenerator`**. As it was defined at a lower level, the user may be trying to override the custom view with an svg for that node in particular. 

In this case, I'm setting the `viewGenerator` property to `null`, so the default custom view is not applied for that node in particular. Since the `Node` component receive both `viewGenerator` and `svg` as props, if both of them has values there's no way to know if the `viewGenerator` property was inherited from the global settings and it should be overriden. 

Maybe I was just overthinking it and all of this logic is unneccessary, but I'd still wanted to explain it before making the change, just in case it made some sense to you and you decide to keep as it is. 🙂

Reply:
But what you're saying wouldn't work, right? That expression means that `viewGenerator` could be assigned a string with the SVG url as a value. This would cause the current code to fail, as the `viewGenerator` property is expected to be a function. 

I could check the type of the `viewGenerator` property inside of `render` and assume it's an svg url if it's not a function, but I'm not really into the idea. That would be pretty hard to understand, as we already have a separate property for the `svg` and it makes no sense to have a `viewGenerator` property which behaves as both. 

I think I'm going to do this instead:

```
viewGenerator: node.viewGenerator || config.node.viewGenerator,
overrideViewGenerator: !node.viewGenerator && node.svg
```

This is far more readable than what I had before, and it should work just fine. What do you think?

Reply:
Ok forget the `node.viewGenerator || node.svg || config.node.viewGenerator || null` it does not work :smile: , yes that last suggestion of yours shows explicitly that the `node.svg` overrides the view generator maybe rename the override to `overrideGlobalViewGenerator` instead since it overrides the global one?